include/alpaka/math/max/MaxUniformCudaHipBuiltIn.hpp

```diff
@@ -52,6 +50,12 @@ namespace alpaka
                         return ::fmax(x, y);
                     else
                         static_assert(!sizeof(Tx), "Unsupported data type");
+
+                    using Ret [[maybe_unused]] = std::conditional_t<
+                        std::is_integral_v<Tx> && std::is_integral_v<Ty>,
+                        decltype(::max(x, y)),
+                        std::conditional_t<is_decayed_v<Tx, float> && is_decayed_v<Ty, float>, float, double>>;
+                    ALPAKA_UNREACHABLE(Ret{});
```

Comment:
That seems like a lot of magic to just supress a warning. Is there no pragma we can wrap around the entire alpaka headers?

Reply:
Warning pragmas are compier specific, no? So this is more portable.